tests/conftest.py

```diff
@@ -120,20 +113,23 @@ class JiraTestManager(object):
         CI_JIRA_ADMIN (str): Admin user account name.
         CI_JIRA_USER (str): Limited user account name.
         max_retries (int): number of retries to perform for recoverable HTTP errors.
+        initialized (bool): if init was successful.
     """
 
     __shared_state: Dict[Any, Any] = {}
 
-    def __init__(self, jira_hosted_type="Server"):
+    def __init__(self, jira_hosted_type=os.environ.get("CI_JIRA_TYPE", "Server")):
         """Instantiate and populate the JIRA instance"""
         self.__dict__ = self.__shared_state
 
         if not self.__dict__:
             self.initialized = False
             self.max_retries = 5
+            self._cloud_ci = False
```

Comment:
I'll voice again my worries ... shouldn't we have 2 classes where we override the differences between cloud and server?
maybe part of a larger refactoring?
The if'ing seems like something easy forgotten in later updates :-)

Reply:
Definitely agree that this isn't optimal, and probably needs some refactoring down the line.
My intention with the `@allow_on_cloud` decorator is that we could leverage the existing tests and port over key tests to run on both without too much code duplication. So the only time you would to do the if else checks would be when we are supporting both in a test case and there is a key difference between Server and Cloud that needs accounting.